packages/core/src/connectivity/scp-cf/token-accessor.ts

```diff
@@ -24,19 +24,20 @@ import {
 
 async function getClientCredentialsToken(
   service: string | Service,
-  userJwt?: string
+  userJwt?: string | JwtPayload
 ): Promise<ClientCredentialsResponse> {
   const serviceCredentials = resolveService(service).credentials;
 
   let subdomain: string;
   let zoneId: string;
 
   if (userJwt) {
-    const decodedJwt = decodeJwt(userJwt);
-    if (decodedJwt.iss) {
-      subdomain = parseSubdomain(decodedJwt.iss);
+    const jwtPayload =
+      typeof userJwt === 'string' ? decodeJwt(userJwt) : userJwt;
+    if (jwtPayload.iss) {
+      subdomain = parseSubdomain(jwtPayload.iss);
     }
-    zoneId = decodedJwt.zid;
+    zoneId = jwtPayload.zid;
```

Comment:
I have a question that is not directly related to this PR.
We can only get potential `subdomain` and `zoneId`, if `userJwt` is provided.
Otherwise, you'll pass `undefined` in line 44.
It seems these two parameters can be `null` (`subdomain`) and optional (`zoneId`) from the [doc](https://github.wdf.sap.corp/CPSecurity/node-xs2sec/blob/master/doc/TokenFlows.md#client-credentials-flow).
Do we have tests that cover the case when no JWT is provided?